### PR TITLE
🌱 Move topology/mergepatch helper to an internal package

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR moves controllers/topology/mergepatch* to an internal package, so the main topology folder is now cleaner

### CODE REVIEW

File: controllers/topology/internal/mergepatch/mergepatch.go [hunk 1]
```diff
@@ -37,11 +37,11 @@ type mergePatchHelper struct {
 	patch []byte
 }
 
-// newMergePatchHelper will return a patch that yields the modified document when applied to the original document.
+// New will return a patch that yields the modified document when applied to the original document.
 // NOTE: In the case of ClusterTopologyReconciler, original is the current object, modified is the desired object, and
 // the patch returns all the changes required to align current to what is defined in desired; fields not defined in desired
 // are going to be preserved without changes.
-func newMergePatchHelper(original, modified client.Object, c client.Client) (*mergePatchHelper, error) {
+func New(original, modified client.Object, c client.Client) (*helper, error) {
```
Comment (sbueringer) [inline controllers/topology/internal/mergepatch/mergepatch.go:44]: WDYT about calling this `NewHelper` (same as with our `util/patch` helper)

**Review by sbueringer — COMMENTED** [review]
One idea +/- golangci-lint

File: controllers/topology/internal/mergepatch/mergepatch.go (hunk 1 above)
Comment (fabriziopandini) [reply controllers/topology/internal/mergepatch/mergepatch.go]: Fixed!

**Comment by sbueringer** [comment]
Thx!
/lgtm

**Pull request merged**